ibis/omniscidb/tests/conftest.py

```diff
@@ -193,10 +237,56 @@ def temp_database(con, test_data_db: str) -> str:
     -------
     str
     """
-    name = _random_identifier('database')
-    con.create_database(name)
-    try:
-        yield name
-    finally:
-        con.set_database(test_data_db)
-        con.drop_database(name, force=True)
+    with temp_helper(con, kind='database', create=True) as ret_value:
+        yield ret_value
+
+
+@pytest.fixture
+def temp_view(con) -> str:
+    """Return a temporary view name.
+
+    Parameters
+    ----------
+    con : ibis.omniscidb.OmniSciDBClient
+
+    Yields
+    ------
+    name : string
+        Random view name for a temporary usage.
+    """
+    with temp_helper(con, kind='view') as ret_value:
+        yield ret_value
+
+
+@pytest.fixture(scope='function')
+def test_view(con, test_table) -> str:
+    """Create a temporary view.
+
+    Parameters
+    ----------
+    con : ibis.omniscidb.OmniSciDBClient
+
+    Yields
+    -------
+    str
+    """
+    with temp_helper(
+        con, kind='view', create=True, create_kwargs={'expr': test_table}
+    ) as ret_value:
+        yield ret_value
+
+
+@pytest.fixture(scope='function')
```

Comment:
these default to function scope already. However, is there a reason we have function scope? aren't these stateful resources? This is part of the reason you have to jump thru hoops to any kind of testing.

I would prefer if you are refactoring anyhow to treat these as proper resources see: https://docs.pytest.org/en/stable/fixture.html

Reply:
these fixtures creates a temporary object for testing (tables, views, databases, ...), and since we're dropping these objects in some tests (for example in `drop_*`) we can't set scope that wider than `function`